In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [3]:
malaria_dataset = pd.read_csv('../../data/cleaned_malaria_data.csv')
malaria_dataset.head()

COUNTRY_NAME  LATITUDE  LONGITUDE VECTOR_SPECIES_COMPLEX VECTOR_SPECIES  \
0        India   12.9396    79.2397          An. stephensi  Not available   
1        India   10.6760    78.7430          An. stephensi  Not available   
2        India   11.2962    76.9414          An. stephensi  Not available   
3        India   29.9600    78.1600          An. stephensi  Not available   
4        India   27.1666    78.0833          An. stephensi  Not available   

  STAGE  YEAR_START  YEAR_END SAMPLING_METHOD MOSQUITO_NUMBER  \
0    NR         NaN       NaN   Not available   Not available   
1    NR      2007.0    2009.0   Not available   Not available   
2    NR         NaN       NaN   Not available   Not available   
3    NR         NaN       NaN   Not available   Not available   
4    NR         NaN       NaN   Not available   Not available   

  BREEDING_HABITAT       ID_METHOD INVASIVE_STATUS  
0    Not available      Morphology          Native  
1    Not available      Morphology          Native  
2    Not available   Not available          Native  
3    Not available      Morphology          Native  
4    Not available  DNA sequencing          Native

In [22]:
#first I i wanted to get a look at the leading number of cases in malaria from the dataset
#I find it odd that india is the highest 
countries = malaria_dataset["COUNTRY_NAME"].value_counts()
print(f'Number of times each country appears in the dataset: {countries}')

Number of times each country appears in the dataset: COUNTRY_NAME
India                               533
Pakistan                            127
Iran (Islamic Republic of)          124
Ethiopia                            114
Myanmar                              80
Yemen                                48
Sudan                                23
Djibouti                             23
Ghana                                18
Saudi Arabia                         16
Eritrea                              13
Lao People's Democratic Republic     11
Sri Lanka                             7
Thailand                              6
Qatar                                 5
Somalia                               5
Kenya                                 5
Afghanistan                           1
Nigeria                               1
Iraq                                  1
United Arab Emirates                  1
Name: count, dtype: int64


In [6]:
latitudes = malaria_dataset["LATITUDE"].value_counts()
print(f'Number of times each lattitude appears in the dataset: {latitudes}')

Number of times each lattitude appears in the dataset: LATITUDE
28.5810    45
22.5720    25
22.6900    24
11.2962    19
28.5615    18
           ..
14.6670     1
14.6680     1
14.6170     1
14.6330     1
13.0359     1
Name: count, Length: 700, dtype: int64


In [7]:
longitudes = malaria_dataset["LONGITUDE"].value_counts()
print(f'Number of times each lattitude appears in the dataset: {longitudes}')

Number of times each lattitude appears in the dataset: LONGITUDE
77.1730    44
72.8560    25
88.3440    23
73.8170    22
76.9414    19
           ..
95.2510     1
95.4990     1
95.2120     1
98.0660     1
77.5973     1
Name: count, Length: 698, dtype: int64


In [10]:
lat_long  = malaria_dataset.groupby(['LATITUDE', 'LONGITUDE']).size()
sorted_count  = lat_long.sort_values(ascending=False)
top = sorted_count.head(5)
print(top)


LATITUDE  LONGITUDE
28.5810   77.1730      44
22.6900   72.8560      24
11.2962   76.9414      19
28.5615   77.1935      18
28.7174   77.2226      18
dtype: int64


LATITUDE  LONGITUDE
28.5810   77.1730      44 #New Dehli
22.6900   72.8560      24 #Gujarat
11.2962   76.9414      19 #Tamil Nadu 
28.5615   77.1935      18 #New Delhi
28.7174   77.2226      18 #Dehli

In [24]:
top_4_locations = malaria_dataset[malaria_dataset["LATITUDE"].isin([28.5810,22.6900,11.2962,28.5615])]
pd.set_option("display.max_rows", len(top_4_locations))
top_4_locations.head(20)


COUNTRY_NAME  LATITUDE  LONGITUDE VECTOR_SPECIES_COMPLEX VECTOR_SPECIES  \
2          India   11.2962    76.9414          An. stephensi  Not available   
18         India   11.2962    76.9414          An. stephensi  Not available   
23         India   11.2962    76.9414          An. stephensi  Not available   
24         India   11.2962    76.9414          An. stephensi  Not available   
25         India   11.2962    76.9414          An. stephensi  Not available   
26         India   11.2962    76.9414          An. stephensi  Not available   
27         India   11.2962    76.9414          An. stephensi  Not available   
28         India   11.2962    76.9414          An. stephensi  Not available   
30         India   11.2962    76.9414          An. stephensi  Not available   
39         India   11.2962    76.9414          An. stephensi  Not available   
40         India   11.2962    76.9414          An. stephensi  Not available   
41         India   11.2962    76.9414          An. stephensi  Not available   
43         India   11.2962    76.9414          An. stephensi  Not available   
44         India   11.2962    76.9414          An. stephensi  Not available   
45         India   11.2962    76.9414          An. stephensi  Not available   
46         India   11.2962    76.9414          An. stephensi  Not available   
187        India   11.2962    76.9414          An. stephensi  Not available   
205        India   11.2962    76.9414          An. stephensi  Not available   
213        India   28.5810    77.1730          An. stephensi  Not available   
214        India   28.5810    77.1730          An. stephensi  Not available   

      STAGE  YEAR_START  YEAR_END       SAMPLING_METHOD MOSQUITO_NUMBER  \
2        NR         NaN       NaN         Not available   Not available   
18       NR         NaN       NaN         Not available   Not available   
23       NR         NaN       NaN         Not available   Not available   
24       NR         NaN       NaN         Not available   Not available   
25       NR         NaN       NaN         Not available   Not available   
26       NR         NaN       NaN         Not available   Not available   
27       NR         NaN       NaN         Not available   Not available   
28       NR         NaN       NaN         Not available   Not available   
30       NR         NaN       NaN         Not available   Not available   
39       NR         NaN       NaN         Not available   Not available   
40       NR         NaN       NaN         Not available   Not available   
41       NR         NaN       NaN         Not available   Not available   
43       NR         NaN       NaN         Not available   Not available   
44       NR         NaN       NaN         Not available   Not available   
45       NR         NaN       NaN         Not available   Not available   
46       NR         NaN       NaN         Not available   Not available   
187      NR         NaN       NaN         Not available   Not available   
205      NR         NaN       NaN         Not available   Not available   
213  ADULTS      1994.0    1994.0  House resting inside   Not available   
214  ADULTS      1994.0    1994.0  House resting inside   Not available   

    BREEDING_HABITAT      ID_METHOD INVASIVE_STATUS  
2      Not available  Not available          Native  
18     Not available     Morphology          Native  
23     Not available     Morphology          Native  
24     Not available     Morphology          Native  
25     Not available     Morphology          Native  
26     Not available     Morphology          Native  
27     Not available     Morphology          Native  
28     Not available     Morphology          Native  
30     Not available     Morphology          Native  
39     Not available     Morphology          Native  
40     Not available     Morphology          Native  
41     Not available     Morphology          Native  
43     Not available     Morphology          Native  
44     Not a

In [35]:
nd1 = top_4_locations[top_4_locations['LATITUDE'] == 28.5810 ]
gj = top_4_locations[top_4_locations['LATITUDE'] == 22.6900]
tml = top_4_locations[top_4_locations['LATITUDE'] == 11.2962]
nd2 = top_4_locations[top_4_locations['LATITUDE'] == 28.5615]

